src/plugins/chatview/index.js

```diff
@@ -65,6 +65,14 @@ converse.plugins.add('converse-chatview', {
         Object.assign(_converse, exports); // DEPRECATED
         Object.assign(_converse.exports, exports);
 
+        if ('registerProtocolHandler' in navigator) {
+        try {
+            const handlerUrl = `${window.location.origin}${window.location.pathname}#converse/chat?jid=%s`;
```

Comment:
On second thought, given that XEP-0147 handles more use-cases than just opening a chat with a JID, I think we should perhaps create another URL with its own handler.

Something like ``${window.location.origin}${window.location.pathname}#converse/action?uri=%s``

And then instead of `routeToChat`, the handler could be called `routeToQueryAction` (because XEP-0147 uses the terminology "query actions").

Reply:
I have create the function under utils file of the chatview plugin

Reply:
It would be great if you could also write tests for the work you've done, so that we have more confidence that it's correct and so that future regressions are caught.

To do so, you can create a new file here: https://github.com/conversejs/converse.js/tree/master/src/plugins/chatview/tests

You can call it `route-actions.js`.

I'm not 100% sure what the best approach will be to write the tests, perhaps just calling `routeToQueryActions` manually and then confirming what happens.

If you get stuck or need help, please ask and I'll see if I can help you.